src/uct/ib/ud/base/ud_iface.c

```diff
@@ -137,30 +161,48 @@ uct_ud_ep_t *uct_ud_iface_cep_get_ep(uct_ud_iface_t *iface,
     }
 
     conn_match = ucs_conn_match_get_elem(&iface->conn_match_ctx, peer_address,
-                                         conn_sn, queue_type, is_private);
+                                         conn_sn, queue_type, is_rx_only);
     if (conn_match == NULL) {
         return NULL;
     }
 
     ep = ucs_container_of(conn_match, uct_ud_ep_t, conn_match);
     ucs_assert(ep->flags & UCT_UD_EP_FLAG_ON_CEP);
 
-    if (is_private) {
+    if (is_rx_only) {
+        ucs_assertv((ep->flags & UCT_UD_EP_FLAG_TX_RX) == UCT_UD_EP_FLAG_RX,
+                    "ep=%p flags=0x%x", ep, ep->flags);
         ep->flags &= ~UCT_UD_EP_FLAG_ON_CEP;
+    } else {
+        ucs_assertv(ep->flags & UCT_UD_EP_FLAG_TX_RX, "ep=%p flags=0x%x", ep,
+                    ep->flags);
     }
 
     return ep;
 }
 
+int uct_ud_iface_cep_has_ep(uct_ud_ep_t *ep)
+{
+    int res = (ep->flags & UCT_UD_EP_FLAG_TX_RX) &&
+              !(ep->flags & UCT_UD_EP_FLAG_CONNECT_TO_EP);
+
+    ucs_assertv(res == !!(ep->flags & UCT_UD_EP_FLAG_ON_CEP),
+                "ep=%p flags=0x%x res=%d", ep, ep->flags, res);
+
+    return res;
+}
```

Comment:
seems redundant, could just check UCT_UD_EP_FLAG_ON_CEP

Reply:
I'm going to remove `UCT_UD_EP_FLAG_ON_CEP` flag in favour of TX/RX flags which mean that endpoint is in CEP.
I kept it for debugging reason and check that all connection matching cases are covered.